src/core/plugin/luapi_application.h

```diff
@@ -218,6 +220,48 @@ static int applib_msgbox(lua_State* L) {
     return 1;
 }
 
+/**
+ * Example 1: app.openDialog("Test123", {[1] = "Yes", [2] = "No"}, "cb")
+ * Pops up a message box with two buttons "Yes" and "No" and executed function "cb" whose single parameter is the number
+ * corresponding to the button the user clicked.
+ * Warning: the callback function is never called if the dialog is closed without pressing one of the custom buttons.
+ */
+static int applib_openDialog(lua_State* L) {
+    const char* msg = luaL_checkstring(L, 1);
+
+    // discard any extra arguments passed in
+    lua_settop(L, 3);
+    luaL_checktype(L, 2, LUA_TTABLE);
+
+    lua_pushnil(L);  // initial key for table traversal with `next`
+
+    std::vector<XojMsgBox::Button> buttons;
+
+    while (lua_next(L, 2) != 0) {
+        int index = static_cast<int>(lua_tointeger(L, -2));
+        const char* buttonText = luaL_checkstring(L, -1);
+        lua_pop(L, 1);
+
+        buttons.emplace_back(buttonText, index);
+    }
+
+    std::string cb = luaL_optstring(L, 3, "");
+    g_message("applib_openDialog: Callback name: %s", cb.c_str());
+    Plugin* plugin = Plugin::getPluginFromLua(L);
+
+    const std::string& pluginName = plugin->getName();
+    bool error = false;
```

Comment:
We should probably add an optional parameter `error` to `openDialog`, which defaults to false.